### PR TITLE
Introduce query API handler and query parser

### PR DESCRIPTION
**Pull request by badiib**

### CODE REVIEW

File: cmd/query/app/handler.go [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+	structuredRes := structuredResponse{
+		Data:  services,
+		Total: len(services),
+	}
+	aH.writeJSON(w, &structuredRes)
+}
+
+func (aH *APIHandler) getOperationsLegacy(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	service := vars[serviceParam] //given how getOperationsLegacy is used, this will never work never be empty
```
Comment (yurishkuro) [inline cmd/query/app/handler.go]: not sue I follow the comment. And why is this called Legacy?

File: cmd/query/app/handler.go [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+	}
+}
+
+// RegisterRoutes registers routes for this handler on the given router
+func (aH *APIHandler) RegisterRoutes(router *mux.Router) {
+	// TODO why single-trace resource below accept POST method?
+	router.HandleFunc(fmt.Sprintf("/%s/traces/{%s}", aH.httpPrefix, traceIDParam), aH.getTrace).Methods(http.MethodPost, http.MethodGet)
+	router.HandleFunc(fmt.Sprintf(`/%s/traces`, aH.httpPrefix), aH.search).Methods(http.MethodGet)
+	router.HandleFunc(fmt.Sprintf(`/%s/services`, aH.httpPrefix), aH.getServices).Methods(http.MethodGet)
+	// TOOD - remove this when UI catches up
```
Comment (yurishkuro) [inline cmd/query/app/handler.go]: this comment should be on the next line?

File: cmd/query/app/handler.go (hunk 1 above)
Comment (badiib) [inline cmd/query/app/handler.go]: Legacy is the function that is called for endpoints like:

api/services/{someServiceName}/operations.

We agreed to decommission this kind of operation name lookup endpoint.

Comment (yurishkuro) [inline cmd/query/app/handler.go]: the comment's wording "will never work never be empty"

**Pull request merged**